### ISSUE TITLE
igx-grid - calculation of grid height has been broken when enabling/disabling filter row after updating to v7.0.4.

### ISSUE THREAD

**Issue by tkiryu** [IgniteUI/igniteui-angular#3363]
### Description  
I checked the fix for #3255 and found that calculation of grid height has been broken when enabling/disabling filter row after updating to v7.0.4. 

 * igniteui-angular version: 7.0.4
 * browser: All

### Steps to reproduce  
Same as #3255.
1. Run the attached sample.
2. Click "Enable Filtering" button.
3. Click "Filter" icon on any column.
4. Click "Disable Filtering" button.

### Result  
- At step 1
There is no problem.
![image](https://user-images.githubusercontent.com/15909409/49717810-941f2680-fc9b-11e8-9b04-d926396250e0.png)

- After step 2
The body height of grid shrinks.
![image](https://user-images.githubusercontent.com/15909409/49717787-7e116600-fc9b-11e8-8919-110c6eb0fd07.png)
- After step 4
The body of grid gets zero height.
![image](https://user-images.githubusercontent.com/15909409/49717840-b022c800-fc9b-11e8-9136-6d0e5096f8e1.png)

### Expected result  
The body height of grid should not shrink as before.

### Attachments  
Same as #3255.
[igx-grid.zip](https://github.com/IgniteUI/igniteui-angular/files/2637917/igx-grid.zip)

**Comment by bkulov** [comment]
This could also be related to #3290

**Comment by bkulov** [comment]
@tkiryu The sample you've attached is using Angular 6 and IgniteUI for Angular 6.2.0. In that version the bug you've described is not reproducible. 
The bug you've described is reproducible in IgniteuUI for Angular 7.0.4 but it's not related to the filtering. Filtering itself is working properly when enabling/disabling it in the grid. 
The actual problem here is that when there is no height set on the grid (which is the case in this bug), the grid tries to calculate it.

I'll change the title of this issue a little bit and will reassign this issue to the people who wrote the height logic.

I'm also attaching the proper sample converted to IgniteUI 7.0.4.
[igx-grid.zip](https://github.com/IgniteUI/igniteui-angular/files/2663174/igx-grid.zip)

**Comment by bkulov** [comment]
@tkiryu There is something else wrong with the sample as the grid doesn't have any styles. Not sure if this affects the current behavior.

**Comment by tkiryu** [comment]
@bkulov 
> The sample you've attached is using Angular 6 and IgniteUI for Angular 6.2.0.

Oh, I'm sorry about that. 

> The bug you've described is reproducible in IgniteuUI for Angular 7.0.4 but it's not related to the filtering. Filtering itself is working properly when enabling/disabling it in the grid.
The actual problem here is that when there is no height set on the grid (which is the case in this bug), the grid tries to calculate it.

Thank you for investigating. I get your point.

I think calculation of grid height is fragile. I update to the latest igniteui-angular as soon as it is released and often encounter such an issue. I hope that the height calculation logic will be more stable.

**Comment by bkulov** [comment]
@tkiryu I've assigned the bug to @ChronosSF who has more knowledge in the area.

**Comment by tkiryu** [comment]
@bkulov , thanks!

**Comment by tkiryu** [comment]
When `igx-grid` doesn't have scrollbar, this issue occurs.

When `igx-grid` has scrollbar, this issue won't occur.

**Comment by kdinev** [comment]
@tkiryu Something seems quite wrong with the theming in these screenshots. Is this from the theming, or is it the sample? 

The grid height calculations are based on the record heights, which are considered before the grid is rendered, because virtualization doesn't work if content is already rendered. The record heights seem unnatural in this sample, even for compact display density, so this may be affecting your rendering. Again, I think this is related to the theme.

**Comment by MayaKirova** [comment]
Duplicate of https://github.com/IgniteUI/igniteui-angular/issues/4458, which is already fixed.
